### DISCUSSION TITLE
Someone wants a blog archive as dropdown list?

### DISCUSSION THREAD

**Discussion by schulle4u** [datenstrom/community#260] [Ask a question] [answered]
WordPress has an archive widget which can be configured to select the months from a dropdown list. This especialy can be useful for blogs with lots of articles during many years, where a long list of months could heavily blow up the sidebar. Though my blog never will fall in this category, I'd like to have a handy dropdown list as an optional argument for the `[blogarchive]` shortcut. 

As far as I can see, the existing code could easily be modified with an extra loop to parse the list of months into a dropdown list. The shortcut needs to be extended with an optional `style` argument, which, if not given, will show the months in a list as usual. Example: 

```
## Archive
[blogarchive /blog/ dropdown]
```

I can try to modify the blog plugin and make a pull request if more people than me think this is useful. 

Steffen

**Comment by markseuffert** [comment, marked as answer]
As a simple alternative you can place the `[blogarchive]` shortcut into a new page. Where you would like to show the dropdown list, you make a link to the new page. Your visitors can now click the link and see the archive with all months. Tested with blog plugin 0.7.2. 

Here's an example content file: 

```
---
Title: Archive
---
[blogarchive /blog/ 0]
```

Just an idea. 😄

**Comment by schulle4u** [comment]
Thanks, thats indeed a very good solution as well and will avoid a bit of headache. I should have come to this conclusion already, because I found a similar solution for blog tags. In your example you set the `pagesmax` parameter to 0, I guess thats meant to show all available items? I did this by setting a high value. 

Sidebar:
```
## Tags
[All Tags](/blog/tags)  
[blogtags /blog/ 5]

```

Page content:
```
---
Title: All Tags
---
List of all tags.

[blogtags /blog/ 100]
```

Must have been the aftermath of WordPress which prevented me from using Yellows simplicity. ;)

Steffen

**Comment by markseuffert** [comment]
Yes, 0 means unlimited pages... any high value works as well.

Have a look at Al Garcia's blog, he uses this idea:

http://notas.litelate.com
